src/app/myworkflows/myworkflows.service.ts

```diff
@@ -99,7 +99,18 @@ export class MyWorkflowsService extends MyEntriesService<Workflow, OrgWorkflowOb
 
   registerEntry(entryType: EntryType | null) {
     if (entryType === EntryType.BioWorkflow) {
-      this.matDialog.open(RegisterWorkflowModalComponent, { width: '600px' });
+      const dialogRef = this.matDialog.open(RegisterWorkflowModalComponent, { width: '600px' });
+      dialogRef
+        .afterClosed()
+        .pipe(takeUntil(this.ngUnsubscribe))
```

Comment:
This service is a singleton, it's never destroyed. So this subscription is never closed. I believe this causes a memory leak. In general, there should never be subscriptions within services (except the HTTP ones).

Reply:
Apparently, it auto closes: https://stackoverflow.com/a/54923792

Just remove the pipe take until.  also remove `extends Base` from the service as it's a trap